source/Taboos/FiniteSubsetTaboo.lagda

```diff
@@ -0,0 +1,193 @@
+---
+title:        Kuratowski-finite Subset Taboo
+author:       Ayberk Tosun
+date-started: 2023-11-27
+---
+
+Based on proof that I learned from Martín Escardó on 2023-11-21, though the
+proof I ended up writing here ended up being a bit different.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.PropTrunc
+open import UF.FunExt
+
+module Taboos.FiniteSubsetTaboo (pt : propositional-truncations-exist)
+                                (fe : Fun-Ext) where
+
+open import Fin.Kuratowski pt
+open import MLTT.Spartan
+open import MLTT.Negation
+open import Naturals.Order
+open import UF.Sets
+open import Fin.Type
+open import UF.Powerset
+open import UF.Powerset-Fin pt
+open import UF.SubtypeClassifier
+open import UF.Logic
+open import UF.DiscreteAndSeparated
+open import UF.ImageAndSurjection pt
+open import UF.Subsingletons
+open import UF.ExcludedMiddle
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier-Properties using (Ω-is-set)
+open PropositionalSubsetInclusionNotation fe
+
+open AllCombinators pt fe
+
+open PropositionalTruncation pt
+
+\end{code}
+
+Let us first define a propositional version of `is-Kuratowski-finite-subset` for
+the sake of convenience.
+
+\begin{code}
+
+is-Kuratowski-finite-subsetₚ : {X : 𝓤  ̇} → 𝓟 X → Ω 𝓤
+is-Kuratowski-finite-subsetₚ P =
+ is-Kuratowski-finite-subset P , being-Kuratowski-finite-is-prop
+
+\end{code}
+
+We now define a predicate expressing the taboo we are interested in: given a
+type `X`, `has-finite-subset-property X` expresses that for every
+Kuratowski-finite subset `F ⊆ X`, any further subset of `S ⊆ F` is also
+Kuratowski-finite.
+
+\begin{code}
+
+finite-subset-property : (X : 𝓤  ̇) → Ω (𝓤 ⁺)
```

Comment:
Maybe this could be called `Kuratowski-finiteness-is-hereditary` instead.

Reply:
Done.